pinot-spi/src/main/java/org/apache/pinot/spi/config/provider/PinotConfigProvider.java

```diff
@@ -57,4 +58,19 @@ public interface PinotConfigProvider {
    *         registered.
    */
   boolean registerSchemaChangeListener(SchemaChangeListener schemaChangeListener);
+
+  /**
+   * Returns the logical table for the given logical table name.
+   * @param logicalTableName the name of the logical table
+   * @return the logical table
+   */
+  LogicalTableConfig getLogicalTable(String logicalTableName);
```

Comment:
(minor) Same for other places where config is returned
```suggestion
  LogicalTableConfig getLogicalTableConfig(String logicalTableName);
```